cloudflare/data_source_zone.go

```diff
@@ -0,0 +1,109 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+func dataSourceCloudflareZone() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"account_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"paused": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"plan": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name_servers": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"vanity_name_servers": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareZoneRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Reading Zones")
+	client := meta.(*cloudflare.API)
+
+	zoneID, zoneIDExists := d.GetOk("zone_id")
+	zoneID = zoneID.(string)
+
+	name, nameExists := d.GetOk("name")
+	name = name.(string)
+
+	if nameExists && zoneIDExists {
+		return fmt.Errorf("zone_id and name arguments can't be used together")
```

Comment:
heh, you're actually already preventing zone ID and name from being used together like i suggested further down. might just need to update the documentation to reflect this.